core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java

```diff
@@ -270,6 +270,11 @@
           null,
           OperandTypes.EXACT_NUMERIC_EXACT_NUMERIC);
 
+  /**
+   * Casting operator used by PostgreSQL '<code>::</code>'.
+   */
+  public static final SqlBinaryOperator PG_CAST = new PostgreSQLCastOperator();
+
```

Comment:
The new operator is to have a special unparsing logic so when this operator is unparsed, it's unparsed as `OP1 :: Type` instead of `CAST(OP1 AS Type`.
I believe it's ok to unparse it as a `CAST` function but I'll still have to parse this special binary operator and produce a function insted.

Reply:
It seems that the `::` operator is equivalent to `CAST` operator [1], so the we only need to do 2 things:

1. support expression parse for `::`
2. customize the unparse for `CAST` operator for `POSTGRES` dialect, but because it's equivalent, unparse to `::` seems not that necessary.

[1] https://www.postgresql.org/docs/8.1/sql-syntax.html

Reply:
Understood, but what if the query wasn't originally written this way ? Besides, parsing a binary operator and producing a function isn't easy and it would complicate this part of the parser for not much gain.